### ISSUE TITLE
Lesson 2: Data cleaning and production; SGD from scratch

### ISSUE THREAD

**Issue by datalass1** [datalass1/fastai#21]
We start today's lesson learning how to build your own image classification model using your own data, including topics such as:

- Image collection
- Parallel downloading
- Creating a validation set, and
- Data cleaning, using the model to help us find data problems.

I'll demonstrate all these steps as I create a model that can take on the vital task of differentiating teddy bears from grizzly bears. Once we've got our data set in order, we'll then learn how to productionize our teddy-finder, and make it available online.

In the second half of the lesson we'll train a simple model from scratch, creating our own gradient descent loop. In the process, we'll be learning lots of new jargon, so be sure you've got a good place to take notes, since we'll be referring to this new terminology throughout the course (and there will be lots more introduced in every lesson from here on).

**Comment by datalass1** [comment]
### Example Projects and plan for Lesson 2 (0-15mins)
Lots of interesting project examples, what will we learn and learn by doing motivation!

### Lets get some teddy bear datasets (15mins
#### Step1: 
Search the images on [Google Image](https://www.google.com/imghp?hl=EN)
#### Step2:
Type Ctrl+Shift+J and paste: 
urls = Array.from(document.querySelectorAll('.rg_di .rg_meta')).map(el=>JSON.parse(el.textContent).ou);
window.open('data:text/csv;charset=utf-8,' + escape(urls.join('\n')));

Change name of the download file of urls to match the image class with .txt file type. 

#### Step3:
Create directories for the imagery data on the GCP next to the lesson notebooks.
```
path = Path('data/bears')
dest = path/folder
dest.mkdir(parents=True, exist_ok=True)
```
#### Step4:
Scp the data (txt files) onto the GCP server
```
$ (base) ubuntu@ubuntu:~/git/fastai$ gcloud compute scp ~/Downloads/urls_grizzly.txt jupyter@my-fastai-instance:~/tutorials/fastai/course-v3/nbs/dl1/data/bears/grizzly/ 

$ (base) ubuntu@ubuntu:~/git/fastai$ gcloud compute scp ~/Downloads/urls_black.txt jupyter@my-fastai-instance:~/tutorials/fastai/course-v3/nbs/dl1/data/bears/black 
   
$ (base) ubuntu@ubuntu:~/git/fastai$ gcloud compute scp ~/Downloads/urls_teddys.txt jupyter@my-fastai-instance:~/tutorials/fastai/course-v3/nbs/dl1/data/bears/teddys

```
